### PR TITLE
fix #25

### LINKED ISSUE #25: Show more definitions
It's only showing the first info on the site.  If you look at https://dictionary.cambridge.org/dictionary/english/stick you will see about 10 definitions.

And this is the cli output:

```
stick (noun)
A thin piece of wood or other material.

Examples:
• The old man was carrying a load of sticks.
• Police said that the child had been beaten with a stick.
```

### PR DESCRIPTION
**Pull request by xerenahmed**
fix #25

- fix showing missing definitions
- shows more info about the word (B1, [T] etc)
- shows language (UK/US) and has an option "--src/-s" that accepts "uk/us/both" (default is both)
Before
```

consider (verb)
To spend time thinking about a possibility or making a decision.

Examples:
• Don't make any decisions before you've considered the situation.
• [ + question word ] Have you considered what you'll do if you don't get the job?

=====*=====

consider (verb)
To think about a particular subject or thing or about doing something or about whether to do something.

Examples:
• Consider Clara Barton, who founded the American Red Cross.
• We considered moving to California, but decided not to.

```

After:
```

consider (verb) UK B1 [ I or T ]
To spend time thinking about a possibility or making a decision.

Examples:
• Don't make any decisions before you've considered the situation.
• [ + question word ] Have you considered what you'll do if you don't get the job?

=====*=====

consider (verb) UK C1 [ T ]
To give attention to a particular subject or fact when judging something else.

Examples:
• You've got to consider the time element when planning the whole project.
• [ + question word ] If you consider how long he's been learning the piano, he's not very good.

=====*=====

consider (verb) UK C2 [ T ]
To care about or respect other people or their feelings and wishes.

Examples:
• Did you consider your mother and how she's going to feel about you leaving?
• She never considers anyone but herself - she's totally selfish!

=====*=====

consider (verb) UK B2 [ T often + obj + (to be) + noun/adj ]
To believe someone or something to be, or think of him, her, or it as something.

Examples:
• He is currently considered (to be) the best British athlete.
• We don't consider her to be right for the job.

=====*=====

consider (verb) US
To think about a particular subject or thing or about doing something or about whether to do something.

Examples:
• Consider Clara Barton, who founded the American Red Cross.
• We considered moving to California, but decided not to.

=====*=====

consider (verb) US
To care about or respect.

Examples:
• Before raising the admission prices, consider the fans.

=====*=====

consider (verb) US
To believe to be; to think of as.

Examples:
• What some people would consider a personal attack, Andy considers a friendly discussion.

```

Warning: an error occurs if the cache is from old code. Because the cache does not have "Info" and src data.

### CODE REVIEW

**Comment by rocktimsaikia** [comment]
Personally I only need the UK definitions. But I guess there can be a option to show the US definitions too ie, `--dictionary` and it should default to `UK`.

I don't want the output to be verbose and hard to read. We can add the which dictionary it is sourcing from ie, UK/US; but get rid of the other infos like `B2 [ T often + obj + (to be) + noun/adj ]`. The definition, word type and dictionary source is enough info already.

This should be the final outout for the header:
```sh
consider (verb) US
```

**Comment by xerenahmed** [comment]
Well, the other info can be useful but it would be hidden if an option is not used like `--describe`

File: cambd/cambd.py [hunk 1]
```diff
@@ -66,46 +66,48 @@ def get_definitions(word: str):
...
-    containers = soup.find_all(attrs={"class": "entry-body__el"})
-    for div in containers:
-        word_type = div.find(attrs={"class": "dpos"})
-        word_type = word_type.get_text() if word_type is not None else None
-
-        # no word_type means this definition is past/past-particle form of the word
-        # recurs with the original word
-        if word_type is None:
-            original_word = div.find(attrs={"class": "dx-h"}).get_text()
-            return get_definitions(original_word)
```
Comment (rocktimsaikia) [inline cambd/cambd.py:79]: Why remove this? This is required to handle past and particle word lookup.

File: cambd/cambd.py [hunk 2]
```diff
@@ -66,46 +66,48 @@ def get_definitions(word: str):
...
+    for container in containers:
+        dcons = container.find_all(attrs={"class": "dsense"})
+        dictionary_title = container.find(attrs={"class": "region"}).get_text().upper()
+
+        for dcon in dcons:
+            word_type = dcon.find(attrs={"class": "dsense_pos"})
+            word_type = word_type.get_text() if word_type is not None else None
+
+            def_info = dcon.find(attrs={"class": "def-info"})
+            def_info = def_info.get_text() if def_info is not None else None
```
Comment (rocktimsaikia) [inline cambd/cambd.py]: I have never used this info tbh. It just makes it more verbose. Lets remove this for now.

**Review by rocktimsaikia — CHANGES_REQUESTED** [review]

File: cambd/cambd.py (hunk 1 above)
Comment (xerenahmed) [reply cambd/cambd.py:79]: In my reproduces I didn't see .dx-h element. And it made some errors when word_type is None. Can you send me a valid word for that? So I can handle.

File: cambd/cambd.py (hunk 2 above)
Comment (xerenahmed) [reply cambd/cambd.py]: I don't use [ T + I noun] someting but B2 C1 infos tells me how it is common or rare in usage. I found this useful when learning english though.

File: cambd/cambd.py (hunk 1 above)
Comment (rocktimsaikia) [reply cambd/cambd.py:79]: Try `nuked` with the current live version of cambd. And try it without this block of code, You will see the diff

Comment (xerenahmed) [reply cambd/cambd.py:79]:
```
nuked (None) UK
Past simple and past participle of nuke.

Examples:

=====*=====

nuked (None) UK   informal
To bomb somewhere with nuclear weapons.

Examples:
• The two countries were threatening to nuke each other.
```

This is more significant to understand actually. If you want the "nuked" definition you want to learn that it's past tense of nuke, you didn't want to see nuked means "bomb somewhere"

File: cambd/cambd.py (hunk 2 above)
Comment (xerenahmed) [reply cambd/cambd.py]: Made `-v` `--verbose`flag for this.

**Comment by rocktimsaikia** [comment]
With the addition of the new field in the table. It does not store all the definitions. It will only store one definition by default.

image

It should store all definitions by default including both uk,us so that looking up the same word does not scrap the whole page again. Caching of all definitions should NOT depend on what options are being passed.

**Review by rocktimsaikia — APPROVED** [review]

**Comment by rocktimsaikia** [comment]
Thanks for contributing. Looks good.

**Pull request merged**